framework/src/physics/PhysicsBase.C

```diff
@@ -477,3 +495,241 @@ PhysicsBase::addPetscPairsToPetscOptions(
         *this,
         po);
 }
+
+bool
+PhysicsBase::isVariableFV(const VariableName & var_name) const
+{
+  const auto var = &_problem->getVariable(0, var_name);
+  if (const auto & field_var = dynamic_cast<MooseVariableFieldBase *>(var))
+    return field_var->isFV();
+  return false;
+}
+
+bool
+PhysicsBase::isVariableScalar(const VariableName & var_name) const
+{
+  return _problem->hasScalarVariable(var_name);
+}
+
+bool
+PhysicsBase::shouldCreateVariable(const VariableName & var_name,
+                                  const std::vector<SubdomainName> & blocks,
+                                  const bool error_if_aux)
+{
+  if (!variableExists(var_name, error_if_aux))
+    return true;
+  // check block restriction
+  auto & var = _problem->getVariable(0, var_name);
+  const bool not_block_restricted =
+      (std::find(blocks.begin(), blocks.end(), "ANY_BLOCK_ID") != blocks.end()) ||
+      allMeshBlocks(blocks);
+  if (!var.blockRestricted() || (!not_block_restricted && var.hasBlocks(blocks)))
```

Comment:
If `blocks` does indicate some kind of block restriction but `var.hasBlocks(blocks)` is `true`, it seems like we should still be returning `false` here